capabilities/index/index.go

```diff
@@ -0,0 +1,114 @@
+package index
+
+import (
+	_ "embed"
+	"fmt"
+
+	ipldprime "github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	ipldschema "github.com/ipld/go-ipld-prime/schema"
+	"github.com/storacha/go-libstoracha/capabilities/types"
+	"github.com/storacha/go-ucanto/core/ipld"
+	"github.com/storacha/go-ucanto/core/result/failure"
+	"github.com/storacha/go-ucanto/core/schema"
+	"github.com/storacha/go-ucanto/ucan"
+	"github.com/storacha/go-ucanto/validator"
+)
+
+//go:embed index.ipldsch
+var assertSchema []byte
+
+var assertTS = mustLoadTS()
+
+func mustLoadTS() *ipldschema.TypeSystem {
+	ts, err := ipldprime.LoadSchemaBytes(assertSchema)
+	if err != nil {
+		panic(fmt.Errorf("loading assert schema: %w", err))
+	}
+	return ts
+}
+
+func AddCaveatsType() ipldschema.Type {
+	return assertTS.TypeByName("AddCaveats")
+}
+
+var IndexAbility = "space/index/*"
+
+// Index capability definition
+// This capability can only be delegated (but not invoked) allowing audience to
+// derive any `space/index/` prefixed capability for the space identified by the DID
+// in the `with` field.
+var Index = validator.NewCapability(
+	IndexAbility,
+	schema.DIDString(),
+	schema.Struct[struct{}](nil, nil, types.Converters...),
+	func(claimed, delegated ucan.Capability[struct{}]) failure.Failure {
+		return equalWith(claimed, delegated)
+	},
+)
+
+// AddCaveats represents the arguments for the space/index/add capability
+type AddCaveats struct {
+	// Link is the Content Archive (CAR) containing the `Index`.
+	Index ucan.Link
+}
+
+func (ic AddCaveats) ToIPLD() (datamodel.Node, error) {
+	return ipld.WrapWithRecovery(&ic, AddCaveatsType(), types.Converters...)
+}
+
+var AddAbility = "space/index/add"
+
+var AddCaveatsReader = schema.Struct[AddCaveats](AddCaveatsType(), nil, types.Converters...)
+
```

Comment:
it may be useful to define the `AddOk` type for responses to `space/index/add` invocations. Even if it's empty, it will be used when creating results and receipts, which are generic and need a type.

You can find an example of this in the [http/put](https://github.com/storacha/go-libstoracha/blob/7c2ab44cf0f349f3768d7ba233079960c4512ec8/capabilities/http/put.go#L42-L50) capability.